lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -139,42 +101,35 @@ defmodule LambdaEthereumConsensus.ForkChoice do
       end
 
     persist_store(state)
-    {:noreply, state}
   end
 
-  @impl GenServer
-  def handle_cast({:on_tick, time}, _store) do
+  @spec on_tick(Types.uint64()) :: :ok
+  def on_tick(time) do
     store = fetch_store!()
     %Store{finalized_checkpoint: last_finalized_checkpoint} = store
 
     new_store = Handlers.on_tick(store, time)
     %Store{finalized_checkpoint: new_finalized_checkpoint} = new_store
     prune_old_states(last_finalized_checkpoint.epoch, new_finalized_checkpoint.epoch)
     persist_store(new_store)
-    {:noreply, new_store}
-  end
-
-  @impl GenServer
-  def handle_info(_msg, state) do
-    {:noreply, state}
   end
 
   ##########################
   ### Private Functions
   ##########################
 
-  def prune_old_states(last_finalized_epoch, new_finalized_epoch) do
+  defp prune_old_states(last_finalized_epoch, new_finalized_epoch) do
     if last_finalized_epoch < new_finalized_epoch do
       new_finalized_slot =
         new_finalized_epoch * ChainSpec.get("SLOTS_PER_EPOCH")
 
-      Task.async(StateDb, :prune_states_older_than, [new_finalized_slot])
-      Task.async(BlockDb, :prune_blocks_older_than, [new_finalized_slot])
+      Task.start_link(StateDb, :prune_states_older_than, [new_finalized_slot])
```

Comment:
Again, let's not call start link unless we're statically supervising that task. Let's have a dynamic supervisor for these.

Reply:
Done!